File: app/src/main/java/com/tobiasschuerg/wordclock/values/Effect.java
package com.tobiasschuerg.wordclock.values;

/**
 * Created by Tobias Schürg on 27.11.2016.
 */

public class Effect implements ClockValue {

    private final boolean esIstEnabled;
    private final int     effect;

    public Effect(int effect, boolean esIstEnabled) {
        this.esIstEnabled = esIstEnabled;
        this.effect = effect;
    }

    @Override
    public byte[] toByte() {
        final byte esIst = (byte) (esIstEnabled ? 1 : 0);
        return new byte[]{'E', (byte) effect, esIst, 0};
    }
}

File: app/src/main/java/com/tobiasschuerg/wordclock/values/Effect.kt
package com.tobiasschuerg.wordclock.values

/**
 * Created by Tobias Schürg on 27.11.2016.
 */

data class Effect(
        val effect: Int,
        val esIstEnabled: Boolean
) : ClockValue {

    override fun toByte(): ByteArray {
        val esIst = (if (esIstEnabled) 1 else 0).toByte()
        return byteArrayOf(
                'E'.toByte(),
                effect.toByte(),
                esIst,
                0
        )
    }
}
